File: interfaces_animals_python/model/duck.py
from model.terrestrial import Terrestrial
from model.swimmer import Swimmer
from model.fly import Fly

class Duck(Terrestrial, Swimmer, Fly):
    def terrestrial(self):
        print("El patp puede correr")

    def swimmer(self):
        print("El pato puede nadar")

    def fly(self):
        print("El pato puede volar")

File: animal_interfaces_java/src/main/java/co/edu/university/model/Duck.java
package co.edu.university.model;

public class Duck extends Animal implements Fly, Swimmer, Terrestrial{
    public Duck(String name, int age, String gender) {
        super(name, age, gender);
    }

    @Override
    public boolean isFlying() {
        return true;
    }

    @Override
    public boolean isSwimming() {
        return true;
    }

    @Override
    public boolean isTerrestrial() {
        return true;
    }

    @Override
    public void runSpeed(double speed) {
        System.out.println("Duck run speed: " + speed);
    }

    @Override
    public void swimSpeed(double speed) {
        System.out.println("Duck swim speed: " + speed);
    }

    @Override
    public void flySpeed(double speed) {
        System.out.println("Duck swim speed: " + speed);
    }

}
